src/App.tsx

```diff
@@ -17,18 +66,19 @@
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter filter={filter} setFilter={setFilter} searchQuery={searchQuery} setSearchQuery={setSearchQuery}/>
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading ? <Loader /> : <TodoList todos={filteredTodos} onSelect={handleSelectTodo} />}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo && (
+        <TodoModal todo={selectedTodo} user={user} onClose={() => setSelectedTodo(null)} />
```

Comment:
Ensure that the `TodoModal` component correctly handles the `todo`, `user`, and `onClose` props. Verify that these props are implemented in the `TodoModal` component.